gen3/neural-networks/object-detection/yolo-world/main.py

```diff
@@ -0,0 +1,255 @@
+        alpha = 0.4  # Transparency factor
+        frame = cv2.addWeighted(overlay, alpha, frame, 1 - alpha, 0)
+
+        cv2.rectangle(frame, (x1, y1), (x2, y2), color, 2)
+
+        label_text = (
+            f"{class_names[detection.label]}: {int(detection.confidence * 100)}%"
+        )
+        text_size = cv2.getTextSize(label_text, cv2.FONT_HERSHEY_SIMPLEX, 0.5, 1)[0]
+        text_x, text_y = x1, y1 - 10
+
+        cv2.rectangle(
+            frame,
+            (text_x, text_y - text_size[1] - 5),
+            (text_x + text_size[0] + 5, text_y + 5),
+            color,
+            -1,
+        )
+
+        cv2.putText(
+            frame,
+            label_text,
+            (text_x, text_y),
+            cv2.FONT_HERSHEY_SIMPLEX,
+            0.5,
+            (255, 255, 255),
+            1,
+        )
+
+    return frame
+
+
+def download_model(url, save_path):
+    if not os.path.exists(save_path):
+        print(f"Downloading model from {url}...")
+        response = requests.get(url)
+        if response.status_code == 200:
+            with open(save_path, "wb") as f:
+                f.write(response.content)
+            print(f"Model saved to {save_path}.")
+        else:
+            raise Exception(
+                f"Failed to download model. Status code: {response.status_code}"
+            )
+    else:
+        print(f"Model already exists at {save_path}.")
+
+    return save_path
+
+
+def extract_text_embeddings(class_names):
+    tokenizer = AutoTokenizer.from_pretrained("openai/clip-vit-base-patch32")
+    text = tokenizer(text=class_names, return_tensors="pt", padding=True)
+    text_onnx = text["input_ids"].detach().cpu().numpy().astype(np.int64)
+    attention_mask = (text_onnx != 0).astype(np.int64)
+
+    textual_onnx_model_path = download_model(
+        "https://huggingface.co/jmzzomg/clip-vit-base-patch32-text-onnx/resolve/main/model.onnx",
+        "clip_textual_hf.onnx",
+    )
+
+    session_textual = onnxruntime.InferenceSession(
+        textual_onnx_model_path,
+        providers=[
+            "TensorrtExecutionProvider",
+            "CUDAExecutionProvider",
+            "CPUExecutionProvider",
+        ],
+    )
+    textual_output = session_textual.run(
+        None,
+        {
+            session_textual.get_inputs()[0].name: text_onnx,
+            "attention_mask": attention_mask,
+        },
+    )[0]
+
+    num_padding = MAX_NUM_CLASSES - len(class_names)
+    text_features = np.pad(
+        textual_output, ((0, num_padding), (0, 0)), mode="constant"
+    ).T.reshape(1, 512, MAX_NUM_CLASSES)
+    text_features = (text_features / QUANT_SCALE) + QUANT_ZERO_POINT
+    text_features = text_features.astype("uint8")
+
+    del session_textual
+
+    return text_features
+
+
+def main(args):
+
+    text_features = extract_text_embeddings(args.class_names)
+    device_info = dai.DeviceInfo(args.device_ip)
+    device = dai.Device(device_info)
```

Comment:
This can be consolidated to [this](https://github.com/luxonis/depthai-experiments/blob/041ad223c5c867ecb9f75725caf88d0c17938feb/gen3/neural-networks/basic-example/main.py#L9)